src/Render/Markdown/MarkdownFactory.cs

```diff
@@ -23,6 +23,13 @@ internal static class MarkdownFactory {
 			renderer.Render( doc );
 		}
 
+		public static string RenderToString( MarkdownDocument doc ) {

```

Comment:
Are there memory concerns here? Since files are all done async, this could put all markdown and html in memory at the same time. Maybe a problem for large directories?

Reply:
Good observation. It would be nice if Scriban was more stream based even for args. I think it's fine for now.